src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts

```diff
@@ -359,6 +366,70 @@ export class BuyCrypto extends IEntity {
     return [this.id, update];
   }
 
+  fiatAmlCheck(
+    inputAmountEur: number,
+    minVolume: number,
+    monthlyAmountInEur: number,
+    bankDataUserData: UserData,
+  ): UpdateResult<BuyCrypto> {
+    const update: Partial<BuyCrypto> =
+      this.bankTx.currency === this.bankTx.txCurrency &&
+      this.target.asset.buyable &&
+      this.bankTx.txAmount >= minVolume &&
+      this.user.userData.annualBuyVolume + inputAmountEur < this.user.userData.depositLimit &&
+      bankDataUserData?.id === this.user.userData.id &&
+      this.user.userData.kycStatus === KycStatus.COMPLETED &&
+      this.user.status === UserStatus.ACTIVE &&
+      this.user.userData.status === UserDataStatus.ACTIVE &&
+      this.user.userData.riskState === RiskState.C &&
+      monthlyAmountInEur <= Config.amlCheckMonthlyTradingLimit
+        ? { amlCheck: AmlCheck.PASS }
+        : { amlCheck: AmlCheck.GSHEET };
+
+    Object.assign(this, update);
+
+    return [this.id, update];
+  }
+
+  fillUp(
+    amountInChf: number,
+    amountInEur: number,
+    minVolumeInCurrency: number,
+    minVolumeChf: number,
+  ): UpdateResult<BuyCrypto> {
+    const userBuyFee = this.user.buyFee;
+    const assetFee = Config.buy.fee.get(this.target.asset.feeTier, this.user.userData.accountType);
+    const fee = userBuyFee && userBuyFee < assetFee ? userBuyFee : assetFee;
```

Comment:
This logic already exists in user.service. Maybe move it to user.entity and use it here as well?